executor/builder.go

```diff
@@ -1002,6 +1002,12 @@ func (b *executorBuilder) buildHashJoin(v *plannercore.PhysicalHashJoin) Executo
 		joinType:      v.JoinType,
 		isOuterJoin:   v.JoinType.IsOuterJoin(),
 		innerEstCount: v.Children()[v.InnerChildIdx].StatsCount(),
+		outerHashJoin: v.OuterHashJoin,
+	}
+	// reverse the inner and the outer
+	if e.outerHashJoin {
+		v.InnerChildIdx = 1 - v.InnerChildIdx
```

Comment:
Actually, we'd better move these lines to the plan building phase?

Reply:
May be not. 
The plan building phase just make choose whether to adopt the  outer hash join, and the executing phase reverses the inner and the outer internally. This way does not need to change a lot of code in the plan build phase.

Reply:
Execute phase better not change the physical plan.

Reply:
Other members may not think so according to previous daily meeting when discussing the solutions. 
On the other hand, the current way is implemented and tested. If taking to way to change physical plans at builing plan phase, it needs to rewrite a lot of code.

Reply:
Why does it need to rewrite a lot of code?
Only the `*LogicalJoin.getHashJoin` and the `PhysicalHashJoin.GetCost` function of HashJoin may be affected?

Reply:
Besides, `buildHashJoin ` here and `explain()`